Don't require too much cmake. Python version check is already patched to find
the right version.

Don't install garbage

--- rclcpp/CMakeLists.txt~	2022-05-10 11:37:52.530600944 +0200
+++ rclcpp/CMakeLists.txt	2022-05-10 11:42:54.698137904 +0200
@@ -129,20 +129,6 @@
   src/rclcpp/waitable.cpp
 )
 
-# By default, without the settings below, find_package(Python3) will attempt
-# to find the newest python version it can, and additionally will find the
-# most specific version.  For instance, on a system that has
-# /usr/bin/python3.10, /usr/bin/python3.11, and /usr/bin/python3, it will find
-# /usr/bin/python3.11, even if /usr/bin/python3 points to /usr/bin/python3.10.
-# The behavior we want is to prefer the "system" installed version unless the
-# user specifically tells us othewise through the Python3_EXECUTABLE hint.
-# Setting CMP0094 to NEW means that the search will stop after the first
-# python version is found.  Setting Python3_FIND_UNVERSIONED_NAMES means that
-# the search will prefer /usr/bin/python3 over /usr/bin/python3.11.  And that
-# latter functionality is only available in CMake 3.20 or later, so we need
-# at least that version.
-cmake_policy(SET CMP0094 NEW)
-set(Python3_FIND_UNVERSIONED_NAMES FIRST)
 
 find_package(Python3 REQUIRED COMPONENTS Interpreter)
 
@@ -257,6 +257,7 @@
 install(
   DIRECTORY include/ ${CMAKE_CURRENT_BINARY_DIR}/include/
   DESTINATION include/${PROJECT_NAME}
+  FILES_MATCHING PATTERN "*.hpp" PATTERN "*.h"
 )
 
 if(TEST cppcheck)
--- rclcpp/CMakeLists.txt~	2025-07-31 14:39:49.989944304 +0200
+++ rclcpp/CMakeLists.txt	2025-07-31 14:43:53.813709731 +0200
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.20)
+cmake_minimum_required(VERSION 3.12)
 
 project(rclcpp)
 
